### DISCUSSION TITLE
UnsatisfiableError when I install Deepmd-kit 2.1.1

### DISCUSSION THREAD

**Discussion by LZYUCL** [deepmodeling/deepmd-kit#1761] [Q&A]
**When I install the gpu version of deepmd on my computer, an UnsatisfiableError will appear when I install version 2.1.1, but there is no problem when I install version 2.0.3. How can I solve it? thanks**

(base) gatsby@Gatsby:~$ conda create -n deepmd-gpu deepmd-kit=2.1.1=*gpu libdeepmd=2.1.1=*gpu lammps-dp=2.0.2 horovod -c https://conda.deepmodeling.org
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: |
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

Package mpi conflicts for:
lammps-dp=2.0.2 -> mpich[version='>=3.3.2,<3.4.0a0'] -> mpi==1.0=mpich
libdeepmd==2.1.1[build=*gpu] -> mpich[version='>=3.3.2,<3.4.0a0'] -> mpi==1.0=mpich
horovod -> mpich[version='>=3.3.2,<3.4.0a0'] -> mpi==1.0=mpich

Package libgcc-ng conflicts for:
lammps-dp=2.0.2 -> libgcc-ng[version='>=7.5.0']
lammps-dp=2.0.2 -> fftw[version='>=3.3.9,<4.0a0'] -> libgcc-ng[version='>=7.2.0|>=7.3.0']

Package python conflicts for:
deepmd-kit==2.1.1[build=*gpu] -> dargs[version='>=0.2.6'] -> python[version='>=2.7,<2.8.0a0|>=3.6|>=3.6,<3.7.0a0|>=3.5,<3.6.0a0|>=3.5']
horovod -> python[version='>=3.10,<3.11.0a0|>=3.8,<3.9.0a0|>=3.9,<3.10.0a0|>=3.7,<3.8.0a0']
deepmd-kit==2.1.1[build=*gpu] -> python[version='>=3.10,<3.11.0a0|>=3.8,<3.9.0a0|>=3.9,<3.10.0a0|>=3.7,<3.8.0a0']
lammps-dp=2.0.2 -> pylammps-dp==2.0.2=pyh6e9818e_6 -> python[version='>=3']
horovod -> cloudpickle -> python[version='>=2.7,<2.8.0a0|>=3|>=3.5|>=3.6|>=3.6,<3.7.0a0|>=3.5,<3.6.0a0']

Package nccl conflicts for:
libdeepmd==2.1.1[build=*gpu] -> nccl[version='>=2.11.4.1,<3.0a0|>=2.12.12.1,<3.0a0']
lammps-dp=2.0.2 -> libdeepmd[version='>=2.0.2'] -> nccl[version='>=2.11.4.1,<3.0a0|>=2.12.12.1,<3.0a0']
horovod -> nccl[version='>=2.11.4.1,<3.0a0|>=2.12.12.1,<3.0a0|>=2.8.3.1,<3.0a0']
deepmd-kit==2.1.1[build=*gpu] -> tensorflow=2.9[build=cuda*] -> nccl[version='>=2.11.4.1,<3.0a0|>=2.12.12.1,<3.0a0']

Package libffi conflicts for:
horovod -> python[version='>=3.9,<3.10.0a0'] -> libffi[version='>=3.2.1,<3.3a0|>=3.3,<3.4.0a0']
deepmd-kit==2.1.1[build=*gpu] -> python[version='>=3.8,<3.9.0a0'] -> libffi[version='>=3.2.1,<3.3a0|>=3.3,<3.4.0a0']

Package libopenblas conflicts for:
lammps-dp=2.0.2 -> plumed[version='>=2'] -> libopenblas[version='>=0.3.10,<1.0a0|>=0.3.18,<1.0a0|>=0.3.13,<1.0a0']
deepmd-kit==2.1.1[build=*gpu] -> numpy -> libopenblas[version='>=0.2.20,<0.2.21.0a0|>=0.3.2,<0.3.3.0a0|>=0.3.20,<1.0a0|>=0.3.3,<1.0a0']

Package mpich conflicts for:
lammps-dp=2.0.2 -> pylammps-dp==2.0.2=pyh6e9818e_6 -> mpich
lammps-dp=2.0.2 -> mpich[version='>=3.3.2,<3.4.0a0|>=3.3.2,<4.0a0']

Package mpi4py conflicts for:
lammps-dp=2.0.2 -> pylammps-dp==2.0.2=pyh6e9818e_6 -> mpi4py
horovod -> mpi4py

Package zlib conflicts for:
lammps-dp=2.0.2 -> zlib[version='>=1.2.11,<1.3.0a0']
lammps-dp=2.0.2 -> plumed[version='>=2'] -> zlib[version='>=1.2.12,<1.3.0a0']

Package openssl conflicts for:
libdeepmd==2.1.1[build=*gpu] -> libtensorflow_cc=2.9[build=cuda*] -> openssl[version='>=1.1.1l,<1.1.2a|>=1.1.1o,<1.1.2a']
deepmd-kit==2.1.1[build=*gpu] -> python[version='>=3.8,<3.9.0a0'] -> openssl[version='>=1.0.2o,<1.0.3a|>=1.0.2p,<1.0.3a|>=1.1.1a,<1.1.2a|>=1.1.1b,<1.1.2a|>=1.1.1c,<1.1.2a|>=1.1.1d,<1.1.2a|>=1.1.1e,<1.1.2a|>=1.1.1g,<1.1.2a|>=1.1.1j,<1.1.2a|>=1.1.1k,<1.1.2a|>=1.1.1l,<1.1.2a|>=1.1.1n,<1.1.2a|>=1.1.1o,<1.1.2a|>=1.1.1i,<1.1.2a|>=1.1.1h,<1.1.2a|>=1.1.1m,<1.1.2a']
horovod -> python[version='>=3.9,<3.10.0a0'] -> openssl[version='>=1.0.2o,<1.0.3a|>=1.0.2p,<1.0.3a|>=1.1.1a,<1.1.2a|>=1.1.1b,<1.1.2a|>=1.1.1c,<1.1.2a|>=1.1.1d,<1.1.2a|>=1.1.1e,<1.1.2a|>=1.1.1g,<1.1.2a|>=1.1.1h,<1.1.2a|>=1.1.1i,<1.1.2a|>=1.1.1j,<1.1.2a|>=1.1.1k,<1.1.2a|>=1.1.1l,<1.1.2a|>=1.1.1n,<1.1.2a|>=1.1.1o,<1.1.2a|>=1.1.1m,<1.1.2a']

Package tensorflow conflicts for:
horovod -> tensorflow[version='2.7.*|2.7.*|2.9.*',build='cpu*|cpu*|cuda*|cuda*']
horovod -> tensorflow-estimator -> tensorflow[version='2.7.0|2.7.0|2.7.0|2.7.0|2.7.0|2.7.0|2.7.0|2.7.0|2.7.0|2.9.0',build='cuda101py39h3452394_0|cuda101py37h3452394_0|cuda113py38h5fe7361_0|cuda113py37h5fe7361_0|cpu_py37hc622def_0|cpu_py310h0716a2c_0|cuda102py310h7cc18f4_0|cuda102py38h7cc18f4_0|cuda102py39h7cc18f4_0|cuda116py39h5dcf97d_0|cuda116py38h5dcf97d_0|cpu_py38h0716a2c_0|cuda116py310h5dcf97d_0|cpu_py39h0716a2c_0|cpu_py38hc622def_0|cuda101py38h3452394_0|cpu_py39hc622def_0|cuda113py39h5fe7361_0']

Package libstdcxx-ng conflicts for:
lammps-dp=2.0.2 -> libstdcxx-ng[version='>=7.5.0']
lammps-dp=2.0.2 -> mpich[version='>=3.3.2,<3.4.0a0'] -> libstdcxx-ng[version='>=7.3.0']The following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.31=0
  - feature:|@/linux-64::__glibc==2.31=0
  - deepmd-kit==2.1.1[build=*gpu] -> __glibc[version='>=2.17']
  - deepmd-kit==2.1.1[build=*gpu] -> cudatoolkit[version='>=11.6,<12'] -> __glibc[version='>=2.17,<3.0.a0']
  - horovod -> __cuda
  - horovod -> __glibc[version='>=2.17']
  - horovod -> cudatoolkit[version='>=11.6,<12'] -> __glibc[version='>=2.17,<3.0.a0']
  - libdeepmd==2.1.1[build=*gpu] -> __glibc[version='>=2.17']
  - libdeepmd==2.1.1[build=*gpu] -> cudatoolkit[version='>=11.6,<12'] -> __glibc[version='>=2.17,<3.0.a0']

Your installed version is: 2.31

**Comment by njzjz** [comment]
See my comment in #1362

**Reply by WU2JING** [reply]
I also meet the problem for Deepmd-kit 2.1.3
